bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GCData.java

```diff
@@ -57,4 +60,13 @@ public final class GCData {
 	public float gdipXOffset, gdipYOffset;
 	public int uiState = 0;
 	public boolean focusDrawn;
+
+	int getNativeDeviceZoom() {
+		return nativeDeviceZoom;
+	}
+
+	void setNativeDeviceZoom(int nativeDeviceZoom) {
+		this.nativeDeviceZoom = nativeDeviceZoom;
+		this.deviceZoom = DPIUtil.getZoomForAutoscaleProperty(nativeDeviceZoom);
+	}
```

Comment:
What is the purpose of these methods?
- `getNativeDeviceZoom()` is rather obsolete as the field for which it serves as an accessor is public anyway.
- `setNativeDeviceZoom()` defines some protocol for setting the `nativeDeviceZoom` along with the `deviceZoom`. While this is a good idea, this only really makes sense if the class provides a proper encapsulation and enforces the protocol for setting the values. As both fields are public, this is not the case here. There is one caller of this method in `GC`, but the same logic is reimplenmented in `Control` as that class cannot access this method for visibility reasons. I would expect the responsibility for correctly setting these values to either be at the class' consumer or at the class itself but not scattered across both.